### PR TITLE
chore: op-chain-ops cleanup

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Cleans up the `op-chain-ops` package metadata, unused packages,
and code styling.

The `op-chain-ops/ether` and `op-chain-ops/db` packages are
removed since they are no longer used.

References to the `op-migrate` bin are removed and documentation
is updated.

This PR should follow #6412 and rebase ontop of it to prevent
large conflicts.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6551** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6551" target="_blank">Graphite</a>  👈
    * **PR #6591** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6591" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6551?utm_source=stack-comment).

**Review by tynes — COMMENTED** [review]
Heroic

File: op-chain-ops/cmd/check-l2/main.go [hunk 1]
```diff
@@ -31,59 +32,88 @@ var defaultCrossDomainMessageSender = common.HexToAddress("0x0000000000000000000
...
-					if err := g.Wait(); err != nil {
-						return err
-					}
-				}
-				g.Go(func() error {
-					return checkPredeploy(clients.L2Client, i)
-				})
-			}
+	if err := app.Run(os.Args); err != nil {
+		log.Crit("error indexing state", "err", err)
```
Comment (tynes) [inline op-chain-ops/cmd/check-l2/main.go]: This log is wrong

File: op-chain-ops/contracts/common.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	// EIP1976ImplementationSlot
```
Comment (tynes) [inline op-chain-ops/contracts/common.go]: I think this may be duplicated in the go codebases

File: op-chain-ops/contracts/common.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	EIP1967ImplementationSlot = common.HexToHash("0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc")
+	// EIP1967AdminSlot
+	EIP1967AdminSlot = common.HexToHash("0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103")
+)
+
+// parseAddress will parse a [common.Address] from a [cli.Context] and return
+// an error if the configured address is not correct.
+func parseAddress(ctx *cli.Context, name string) (common.Address, error) {
```
Comment (tynes) [inline op-chain-ops/contracts/common.go:12]: This seems duplicated from op-service

Comment (tynes) [reply op-chain-ops/contracts/common.go:12]: Wait its different

**Comment by tynes** [comment]
I think this could use a rebase

File: op-chain-ops/contracts/common.go (hunk 2 above)
Comment (refcell) [reply op-chain-ops/contracts/common.go]: Good catch - duplicated here
https://github.com/ethereum-optimism/optimism/blob/develop/op-chain-ops/genesis/helpers.go#L25-L28

File: op-chain-ops/Makefile [hunk 4]
```diff
@@ -1,5 +1,10 @@
-op-migrate:
-	go build -o ./bin/op-migrate ./cmd/op-migrate/main.go
+all: check-l2 eof-crawler
```
Comment (tynes) [inline op-chain-ops/Makefile]: we can remove `eof-crawler` here

File: op-chain-ops/Makefile [hunk 5]
```diff
@@ -10,4 +15,4 @@ fuzz:
 	go test -run NOTAREALTEST -v -fuzztime 10s -fuzz=FuzzAliasing ./crossdomain
 	go test -run NOTAREALTEST -v -fuzztime 10s -fuzz=FuzzVersionedNonce ./crossdomain
 
-.PHONY: op-migrate test
+.PHONY: check-l2 eof-crawler test fuzz
```
Comment (tynes) [inline op-chain-ops/Makefile]: eof-crawler is still down here

**Review by tynes — APPROVED** [review]

**Pull request merged**